src/database-sqlite3.cpp

```diff
@@ -295,6 +352,27 @@ Database_SQLite3::~Database_SQLite3()
 	FINALIZE_STATEMENT(m_stmt_end)
 	FINALIZE_STATEMENT(m_stmt_delete)
 
+	// Try to checkpoint the entire WAL before exit.
+	// so that the database file is self-contained. Hopefully, this also
+	// causes the wal file to be removed.
+	// SQLITE_CHECKPOINT_TRUNCATE is only supported since sqlite 3.8 (jan. 2015)
+#ifdef SQLITE_CHECKPOINT_TRUNCATE
+	int rv = sqlite3_wal_checkpoint_v2(m_database, NULL,
+		SQLITE_CHECKPOINT_TRUNCATE, NULL, NULL);
+#else
+	int rv = sqlite3_wal_checkpoint_v2(m_database, NULL,
+		SQLITE_CHECKPOINT_RESTART, NULL, NULL);
+#endif
+	if (rv != SQLITE_OK) {
+		if (rv == SQLITE_BUSY) {
+			infostream << "Failed to checkpoint database:"
+				<< " the database is in use by another process"
+				<< " (but all data is safe!)"
+				<< std::endl;
+		} else {
+			SQLOK(rv, "Failed to checkpoint database")
+		}
+	}
```

Comment:
This can be flattened out to the following:
```C++
	if (rv == SQLITE_BUSY) {
		infostream << "Failed to checkpoint database:"
			<< " the database is in use by another process"
			<< " (but all data is safe!)"
			<< std::endl;
	} else {
		// No need to check if rv is SQLITE_OK, as that's what SQLOK does.
		SQLOK(rv, "Failed to checkpoint database")
	}
```